lib/IRGen/GenBuiltin.cpp

```diff
@@ -1345,5 +1345,30 @@ void irgen::emitBuiltinCall(IRGenFunction &IGF, const BuiltinInfo &Builtin,
     return;
   }
 
+  if (Builtin.ID == BuiltinValueKind::PackCount) {
+    (void) args.claimAll();
+
+    auto metaTy = argTypes[0].getASTType()->castTo<MetatypeType>();
+    auto tupleTy = metaTy->getInstanceType()->castTo<TupleType>();
+
+    llvm::Value *count = llvm::ConstantInt::get(IGF.IGM.SizeTy, 0);
+
+    // Accumulate the count of the pack.
```

Comment:
It seems odd how you have to iterate over the tuple elements here. Can't you get the PackType from the substitution map, in the same way that the SILCombinerBuiltinVisitors does?

Maybe the builtin should not take a tuple at all? A pack of metatypes should be better. You mentioned that this generated some dead instructions, but we could find a way around that.